src/db.rs

```diff
@@ -323,3 +323,334 @@ pub async fn dependencies_by_dataset_select(
 
     Ok(dependencies)
 }
+
+#[cfg(test)]
```

Comment:
@coderabbitai for the `state_update()` function in this file, please create the follow unit test:
- can we update a data product?
- do we get rejected if there is no data product?

Please create a PR for this change.